python/orca/src/bigdl/orca/data/file.py

```diff
@@ -251,3 +251,37 @@ def put_local_dir_to_remote(local_dir, remote_dir):
             remote_dir = remote_dir[len("file://"):]
         from distutils.dir_util import copy_tree
         copy_tree(local_dir, remote_dir)
+
+
+def put_local_dir_tree_to_remote(local_dir, remote_dir):
+    if remote_dir.startswith("hdfs"):  # hdfs://url:port/file_path
+        import pyarrow as pa
+        host_port = remote_dir.split("://")[1].split("/")[0].split(":")
+        classpath = subprocess.Popen(["hadoop", "classpath", "--glob"],
+                                     stdout=subprocess.PIPE).communicate()[0]
+        os.environ["CLASSPATH"] = classpath.decode("utf-8")
+        fs = pa.hdfs.connect(host=host_port[0], port=int(host_port[1]))
+        if not fs.exists(remote_dir):
+            fs.mkdir(remote_dir)
+        cmd = 'hdfs dfs -put -f {}/* {}/'.format(local_dir, remote_dir)
+        process = subprocess.Popen(cmd, shell=True)
```

Comment:
Is it possible to just use pyarrow or hadoop command instead of interleaving them together?

Reply:
using pyarrow to copy hdfs tree is a little complex. will change to use command all the ways.